### ISSUE TITLE
Subscribe extension to `execute cell` action

### ISSUE THREAD

**Issue by kromanenkov** [jupyterlab/jupyterlab#6143]
Hi all,

I'm trying to write a rather simple jupyterlab extension: after pushing the `execute cell` button I want to wait response from the kernel, get the response metadata and show it on notebook's page in some way.

I briefly looked through jypyterlab-extension projects but didn't find any related extensions. Could you please give me a direction to implement this functionality?

**Comment by jasongrout** [comment]
Can you give a clearer idea of what sort of response metadata you are getting and how you want to display it? It may be that you want to write a rendermime, or you might want to listen to the executed signal.

**Comment by kromanenkov** [comment]
I'm waiting for some execution statistics as the response metadata from the custom kernel. Anyway, I'm trying to find an example of jupyterlab extension which is waiting for the "execute_reply" message from the kernel and display some info near the executed cell or maybe in separate tab.

**Comment by jasongrout** [comment]
Are you wanting a separate object to listen to *all* execute_reply messages from a kernel, or do you just want the thing that sent the execute request to be able to act on the execute reply? It sounds like the former - you want an extension that will be notified on execute_reply for anything that is executed from this frontend to this kernel?

**Comment by kromanenkov** [comment]
Yes, you are right: I need an object listening to all execute_reply messages from the kernel during the current notebook session and reacting accordingly.
